components/form-token-field/token.js

```diff
@@ -36,25 +40,39 @@ function Token( {
 
 	const onClick = () => onClickRemove( { value } );
 
+	const transformedValue = displayTransform( value );
+	const termPositionAndCount = sprintf(
+		/* translators: 1: term name, 2: term position in a set of terms, 3: total term set count. */
+		__( '%1$s (%2$s of %3$s)' ),
+		transformedValue,
+		termPosition,
+		termsCount
+	);
+
 	return (
 		<span
 			className={ tokenClasses }
 			onMouseEnter={ onMouseEnter }
 			onMouseLeave={ onMouseLeave }
 			title={ title }
 		>
-			<span className="components-form-token-field__token-text">
-				{ displayTransform( value ) }
+			<span
+				className="components-form-token-field__token-text"
+				id={ `components-form-token-field__token-text-${ instanceId }` }
+			>
+				<span className="screen-reader-text">{ termPositionAndCount }</span>
+				<span aria-hidden="true">{ transformedValue }</span>
 			</span>
 
 			<IconButton
 				className="components-form-token-field__remove-token"
 				icon="dismiss"
 				onClick={ ! disabled && onClick }
-				label={ sprintf( messages.remove, displayTransform( value ) ) }
+				label={ messages.remove }
+				aria-describedby={ `components-form-token-field__token-text-${ instanceId }` }
```

Comment:
Can we move the span element to be children of IconButton (having the same effect)? Avoiding the need to create ids and the aria-describedby reference?

Reply:
Hm no, because moving the span text inside the button would make the whole text be read out in one go. The whole point of the feedback on https://github.com/WordPress/gutenberg/issues/1339#issuecomment-381137013 was to make the term name on the button be announced after some delay. In fact, the content referenced by `aria-describedby` is announced after a brief interval.

> The label text is simplified since a user is ostensibly hearing the button immediately after the text. For users who are just jumping into the button from elsewhere or who forgot the current tag, aria-describedby still announces the tag text after a moment. Long enough to keep from annoying a user familiar with the page, but still be useful for those who are not.